Sender/BatteryDataStream.Test/BatteryDataStreamTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.IO;
+using Xunit;
+
+namespace BatteryDataStream.Test
+{
+    public class BatteryDataStreamTests
+    {
+        IParameterSource parameterSource = new FakeCSVParameterSource();
+        [Fact]
+        public void LoadDataStreaming_FakeCSVParameterSource_ReturnsStubCSV()
+        {
+            //Arrange
+            DataStreaming dataStreaming = new DataStreaming(parameterSource);
+
+            //Assert
+            string filePath = Directory.GetParent("../../../../") + @"/BatteryDataStream/Data/Parameters.csv";
+            string parameters = dataStreaming.Load(filePath);
+
+            //Assert
+            Assert.True(IsFloat(parameters, 1));
+        }
+
+        [Fact]
+        public void StartDataStreaming_FakeCSVParameterSource_ReturnsStubCSV()
+        {
+            //Arrange
+            DataStreaming dataStreaming = new DataStreaming(parameterSource);
+
+            //Act
+            string parameters = dataStreaming.Start(2);
+
+            //Assert
+            Assert.True(IsFloat(parameters, 0));
+        }
+
+        [Fact]
+        public void StartDataStreaming_NoParam_ReturnsException()
+        {
+            //Assert
+            Assert.Throws<ArgumentNullException>(() => new DataStreaming(null));
+        }
+
+        [Fact]
+        public void StartDataStreaming_SetMaxRows()
+        {
+            parameterSource.MaxRows = 15;
+            //Assert
+            Assert.True(parameterSource.MaxRows == 15);
+        }
+
+        private bool IsFloat(string parameters, int index)
```

Comment:
This is a good test of the output formatting. Why not call it `parseOutputFormat`?